coriolis/osmorphing/base.py

```diff
@@ -120,10 +122,11 @@ class BaseLinuxOSMorphingTools(BaseOSMorphingTools):
     _packages = {}
 
     def __init__(self, conn, os_root_dir, os_root_dev, hypervisor,
-                 event_manager, detected_os_info, osmorphing_parameters):
+                 event_manager, detected_os_info, osmorphing_parameters,
+                 operation_timeout):
```

Comment:
This runs the risk of some provider tools which override `__init__` to fail to call `super().__init__`, could you please make `operation_timeout` a kwarg which defaults to `None` or `-1` (which, by convention, we can consider to be no timeout)?

Please ensure the actual command execution calls can handle a `timeout=None` (most notably wsman.py)

Reply:
We can't really consider `None` the global equivalent of "no timeout", because pywinrm doesn't accept None or 0 as timeout, only > 0 allowed. `paramiko` on the other side accepts both.

Reply:
In that case we add an extra check for WSMan to always default to an internal 1h or something if nothing is provided.

Point is we must decide on a reasonable default for this constructor kwarg to not have the provider tools potentially fail for not setting any.

As mentioned in a different comment, we can convene to either have "None" mean "no timeout", or always demand a numeric input (with values <= 0 counting as no timeout), but we need to be consistent about it.